lib/shopify-cli/tunnel.rb

```diff
@@ -120,6 +121,25 @@ def urls
       []
     end
 
+    ##
+    # Returns Boolean if a tunnel is running on a given port
+    #
+    # #### Parameters
+    #
+    # * `port` - port to check
+    #
+    # #### Returns
+    #
+    # * true / false
+    #
+    def running_on?(port)
```

Comment:
As discussed in Slack, I would inline the private methods below and implement this in a more functional manner, e.g.

```ruby
    def running_on?(port)
      extract_port = ->(tunnel) { URI(tunnel.dig(*TUNNEL_ADDRESS_KEYPATH)).port }
      matches_port = ->(occupied_port) { occupied_port == port }
      stats.fetch(TUNNELS_FIELD, []).map(&extract_port).any?(&matches_port)
    rescue
      false
    end
```